src/mono/mono/eglib/gfile.c

```diff
@@ -127,10 +127,20 @@ g_fopen (const gchar *path, const gchar *mode)
 		return NULL;
 
 #ifdef HOST_WIN32
-	if (is_ascii_string (path) && is_ascii_string (mode)) {
-		fp = fopen (path, mode);
+	gchar *path_mod;
+	if (g_path_is_absolute (path)) {
+		path_mod = g_malloc(strlen(path) + 5);
+		strcpy(path_mod, "\\\\?\\");
+        	strcat(path_mod, path);
```

Comment:
Note that this doesn't handle network paths. E.g. `\\server\share` should be `\\?\UNC\server\share`.

Reply:
Thanks! Updated